src/libra_sumcheck.rs

```diff
@@ -0,0 +1,366 @@
+use p3_field::{ExtensionField, Field, PrimeField32};
+use poly::{Fields, MultilinearExtension};
+use sum_check::{SumCheck, interface::SumCheckInterface, primitives::SumCheckProof};
+use transcript::Transcript;
+
+use crate::utils::{
+    ProveLibraInput, build_phase_one_libra_sumcheck_poly, build_phase_two_libra_sumcheck_poly,
+    prepare_phase_two_params,
+};
+
+pub fn prove_libra_sumcheck<F: Field + PrimeField32, E: ExtensionField<F>>(
+    input: ProveLibraInput<'_, F, E>,
+    transcript: &mut Transcript<F, E>,
+) -> (SumCheckProof<F, E>, Vec<E>, Vec<E>, E, E) {
+    let phase_one_poly = build_phase_one_libra_sumcheck_poly(
+        input.mul_ahg,
+        input.add_b_ahg,
+        input.add_c_ahg,
+        input.w_i_plus_one_poly,
+    );
+
+    let claimed_sum = phase_one_poly.sum_over_hypercube();
```

Comment:
we can avoid this computation
1. we should already know the claimed sum, so might be possible to just pass it in
2. if not nice to pass it in, claimed sum should be equal to the sum of the first round poly (another way of getting it)